### ISSUE TITLE
Stack overflow and broken screenshots error

### ISSUE THREAD

**Issue by JcJet** [Salies/steam-screenshot-backup-tool#7]
I've been using this tool to download my big screenshot collection:
https://steamcommunity.com/id/JcJet/
There's 2 interesting bugs.
First of all, there's a stack overflow error when you resume download after it been interrupted, when reading the JSON data.

Since I know that all the data is already downloaded, and my JS skills are 0, i've just removed this part of the code. But then I've stumbled into something even more interesting, why it was interrupded in the first place. The screenshot, while not being deleted or banned, was just lost from the cloud.
https://steamcommunity.com/sharedfiles/filedetails/?id=1164759563
Well, that's just confirmes that backup tool is very much needed :)
I doubt it will reapear later because I had this issue yesterday, and now found out why it happened, so it doesn't look like a temporary server issues.
Currently, it throws 404 exception and the program stops.

**Comment by Salies** [comment]
Hello! First of all, thanks for the feedback.
I've been meaning to update this program for a while: an overall rework, bug fixing, etc.
I'll try to take some of my vacation time this month to finnaly do it - and when I do, I'll be sure to adress this issue.

Just for curiosity's sake: it's probably due to a weird glitch on Steam where it shows the "no screenshots" page when you first try to load any of the pages.

**Comment by JcJet** [comment]
No, it's definitely not the "no screenshots" issue. The screenshot simply isn't there anymore. Rare issue, out of 14k screenshot, one is corrupted. Also, the stack overflow in recursion happen in two places in this case, when reloading a big json file and when there's a big array of screenshots, because this array is called recursively. I had to trim the json file manually to download the screenshots in two steps.
